### PR TITLE
Add PUT endpoint for submitting DOI information.

### LINKED ISSUE #384: As a SD-Submit API user I want to be able to submit DOI information through dedicated endpoint so it would simplify my work
#### Describe the user story

At the moment when creating DOI, either the frontend or the API user has to create a patch to the folder with a pretty complicated request body. The new endpoint would be taking in JSON with Datacite information and using already existing functionalities for DOI info creation and folder update.

#### Tasks:

- [x] https://github.com/CSCfi/metadata-submitter/pull/423
- [x] https://github.com/CSCfi/metadata-submitter-frontend/issues/799

#### Testing
Integration tests on both front and backend.
Manual API testing.

#### Definition of Done
SD Submit API user is able to add Datacite information to her/his submission with a separate endpoint using JSON format.

### PR DESCRIPTION
**Pull request by csc-felipe**
### Description

Add PUT endpoint for submitting DOI information.
Previously, DOI could be added to a folder only with a patch request.
The new endpoint makes it easier to format requests that add or update DOI information.

### Related issues

Closes #384
### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- Added a new endpoint for submitting DOI information

### Testing

- [x] Unit Tests

### CODE REVIEW

**Review by genie9 — CHANGES_REQUESTED** [review]
The new endpoint is simple and clear, good job! This change though needs to be addressed in integration tests. I do not think new tests are needed if old ones would be adjusted to use a new endpoint.

**Comment by genie9** [comment]
A new endpoint needs to be added to swagger doc https://github.com/CSCfi/metadata-submitter/blob/497bf5982e527fe4ea5ad584fce593c293424bfb/docs/specification.yml#L935

File: metadata_backend/api/handlers/folder.py [hunk 1]
```diff
@@ -601,3 +601,34 @@ async def delete_folder(self, req: Request) -> Response:
...
+        folder_id = req.match_info["folderId"]
+        db_client = req.app["db_client"]
+        operator = FolderOperator(db_client)
+
+        await operator.check_folder_exists(folder_id)
+        await self._handle_check_ownership(req, "folders", folder_id)
+
+        folder = await operator.read_folder(folder_id)
+        doi_info = await self._get_data(req)
+        LOG.info(doi_info)
```
Comment (teemukataja) [inline metadata_backend/api/handlers/folder.py:620]:
```suggestion
```
This was probably left here from testing? Can be removed

**Review by teemukataja — APPROVED** [review]
Could we add a new case also to the python integration tests https://github.com/CSCfi/metadata-submitter/blob/feature/publish_doi_metax_folder/tests/integration/run_tests.py

**Comment by csc-felipe** [comment]
Rebased, and removed `dataset` from minimal publication, as it's not required.

**Review by genie9 — APPROVED** [review]

File: tests/integration/run_tests.py [hunk 2]
```diff
@@ -1737,6 +1752,31 @@ async def test_submissions_work(sess, folder_id):
...
+    await post_object_json(sess, "dataset", folder_id, "dataset.json")
+    await put_folder_doi(sess, "doi", folder_id, "test_doi.json")
+    await publish_folder(sess, folder_id)
+
+    async with sess.get(f"{folders_url}/{folder_id}") as resp:
+        LOG.debug(f"Checking that folder {folder_id} was published")
+        res = await resp.json()
+        assert res["folderId"] == folder_id, "expected folder id does not match"
+        assert res["published"] is True, "folder is published, expected False"
+
```
[the remark this answers is not in the document]
Comment (csc-felipe) [reply tests/integration/run_tests.py:1860]: Let's update it when we deprecate the `patch` endpoint.

I removed the `dataset` in the previous push, as it apparently is not currently required.

**Review by teemukataja — APPROVED** [review]

**Pull request merged**